doc/sphinx/practical-tools/coq-commands.rst

```diff
@@ -42,16 +42,52 @@ toplevel with the command ``Coqloop.loop();;``.
 Batch compilation (coqc)
 ------------------------
 
-The ``coqc`` command takes a name *file* as argument. Then it looks for a
-file named *file*.v, and tries to compile it into a
-*file*.vo file (See :ref:`compiled-files`).
-
-.. caution::
-
-   The name *file* should be a regular Coq identifier as defined in Section :ref:`lexical-conventions`.
-   It should contain only letters, digits or underscores (_). For example ``/bar/foo/toto.v`` is valid,
-   but ``/bar/foo/to-to.v`` is not.
-
+The ``coqc`` command compiles a Coq proof script file with a ".v" suffix
+to create a compiled file with a ".vo" suffix.  (See :ref:`compiled-files`.)
+The components of the filename must be valid Coq identifiers as described in
+:ref:`lexical-conventions`; they should contain only letters, digits or
+underscores (_) with a ".v" suffix on the final component.
+For example ``/bar/foo/toto.v`` is valid, but ``/bar/foo/to-to.v`` is not.
+
+We recommend specifying a logical directory name (which is also the module name)
+with the `-R` or the `-Q` options.
+Generally we recommend using provided utilities such as `make` to build Coq projects.
+See :ref:`building_coq_project`.
+
+.. example:: Compiling and loading a single file
+
+   If `foo.v` is in Coq's current directory, you can use `coqc foo.v`
+   to compile it and then `Require foo.` in your script.  But this
+   doesn't scale well for larger projects.
+
+   Generally it's' better to define a new module:
+   To compile `foo.v` as part of a module `Mod1` that is rooted
+   at `.` (i.e. the directory containing `foo.v`), run `coqc -Q . Mod1 foo.v`.
+
+   To make the module available in `CoqIDE`, include the following lines in the
+   `_CoqProject` file (see :ref:`coq_makefile`) in the directory from which you
+   start `CoqIDE`.  For now, you need to close and reload a named
+   script file for `CoqIDE` to pick up the change or restart `CoqIDE`.
+   The project file name is configurable in `Edit / Preferences / Project`.
+
+      .. coqdoc::
+         -R ../empty_dir/mod1 Mod1
```

Comment:
What is `empty_dir` supposed to represent? Wouldn't it be better to just recommend that the `_CoqProject` file be always located at the root of a project and that CoqIDE (or another compatible editor) be always run from the root of the project as well?

Reply:
I removed empty_dir, that was my local test.

Having to start CoqIDE in a specific directory is inconvenient if you start it from an icon on your desktop.  Icons don't allow for changing the directory.  CoqIDE should let you pick the project within its GUI.  I think there are probably some small code tweaks we could make that would make for a better user experience.

Reply:
I agree.